dispatcher/dispatcher.go

```diff
@@ -274,41 +300,81 @@ func (d *IotxDispatcher) dispatchAction(ctx context.Context, chainID uint32, msg
 	if atomic.LoadInt32(&d.shutdown) != 0 {
 		return
 	}
-	d.enqueueEvent(&actionMsg{
-		ctx:     ctx,
-		chainID: chainID,
-		action:  (msg).(*iotextypes.Action),
-	})
+	subscriber := d.subscriber(chainID)
+	if subscriber == nil {
+		log.L().Debug("no subscriber for this chain id, drop the action", zap.Uint32("chain id", chainID))
+		return
+	}
+	d.actionChanLock.Lock()
+	defer d.actionChanLock.Unlock()
+	l := len(d.actionChan)
+	c := cap(d.actionChan)
+	if l < c {
+		d.actionChan <- &actionMsg{
+			ctx:     ctx,
+			chainID: chainID,
+			action:  (msg).(*iotextypes.Action),
+		}
+		l++
+	} else {
+		log.L().Warn("dispatcher action chan is full, drop an event.")
+	}
+	subscriber.ReportFullness(ctx, iotexrpc.MessageType_ACTION, float32(l)/float32(c))
 }
 
-// dispatchBlockCommit adds the passed block message to the news handling queue.
-func (d *IotxDispatcher) dispatchBlockCommit(ctx context.Context, chainID uint32, msg proto.Message) {
+// dispatchBlock adds the passed block message to the news handling queue.
+func (d *IotxDispatcher) dispatchBlock(ctx context.Context, chainID uint32, msg proto.Message) {
 	if atomic.LoadInt32(&d.shutdown) != 0 {
 		return
 	}
-	d.enqueueEvent(&blockMsg{
-		ctx:     ctx,
-		chainID: chainID,
-		block:   (msg).(*iotextypes.Block),
-	})
+	subscriber := d.subscriber(chainID)
+	if subscriber == nil {
+		log.L().Debug("no subscriber for this chain id, drop the block", zap.Uint32("chain id", chainID))
+		return
+	}
+	d.blockChanLock.Lock()
+	defer d.blockChanLock.Unlock()
```

Comment:
this is needed,
both `HandleBroadcast()` and `HandleTell()` call `dispatchBlockCommit` that will send to this channel